### PR TITLE
Use concat to build facts file

### PR DESCRIPTION
**Pull request by treydock**
Right now if I define a facter::fact and then remove it from catalog, the fact is left behind in facts file.  This would ensure facts file only contains facts put in place by Puppet.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -10,7 +10,7 @@ gem 'metadata-json-lint'
 gem 'puppetlabs_spec_helper', '>= 1.1.1'
 gem 'facter', '>= 1.7.0'
 gem 'rspec-puppet'
-gem 'puppet-lint', :git => 'https://github.com/rodjek/puppet-lint.git'
```
Comment (ghoneycutt) [inline Gemfile]: Leave this here and I'll address the style in another PR so it matches the other modules and uses puppet-lint v2

File: .fixtures.yml [hunk 2]
```diff
@@ -3,5 +3,7 @@ fixtures:
     stdlib:
       repo: 'https://github.com/puppetlabs/puppetlabs-stdlib.git'
       ref: '4.6.0'
+    concat:
+      repo: 'https://github.com/puppetlabs/puppetlabs-concat.git'
```
Comment (ghoneycutt) [inline .fixtures.yml]: please specify the version. This is needed to ensure that testing is consistent.

**Comment by ghoneycutt** [comment]
I'm not opposed to using concat, I just don't understand what it gets us that file_line is not providing.

As for design, I often have facts that are not managed by Puppet that are created during a provisioning process and would not want to remove those by default. It sounds like you do want to manage all of those through Puppet and remove them if they are not, which sounds good, but should be optional. We could solve that by using the resources type and purging resources that match a tag.

**Comment by treydock** [comment]
I redid this PR so that there is a `purge_facts` parameter that defaults to `false` to maintain old behavior. When the value is `true` the `concat` module is used to build the facts file so that unmanaged facts get deleted. This is using commit from #63 and I can rebase if #63 is merged first.

**Pull request closed without merging**